src/Enzo/initial/EnzoInitialCloud.hpp

```diff
@@ -23,64 +23,80 @@ class EnzoInitialCloud : public Initial {
 public: // interface
 
   /// Constructor
-  EnzoInitialCloud(int cycle, double time, int subsample_n,
-		   double cloud_radius, double cloud_center_x,
-		   double cloud_center_y, double cloud_center_z,
-		   double density_cloud, double density_wind,
-		   double etot_wind, double eint_wind,
-		   double velocity_wind, double metal_mass_frac,
-		   bool initialize_uniform_bfield,
-		   const double uniform_bfield[3],
-		   int perturb_Nwaves, double perturb_amplitude,
-                   double perturb_min_wavelength,
-                   double perturb_max_wavelength, unsigned int perturb_seed)
+  EnzoInitialCloud(int cycle, double time, ParameterGroup p) noexcept
     : Initial(cycle,time),
-      subsample_n_(subsample_n),
-      cloud_radius_(cloud_radius),
-      cloud_center_x_(cloud_center_x),
-      cloud_center_y_(cloud_center_y),
-      cloud_center_z_(cloud_center_z),
-      density_cloud_(density_cloud),
-      density_wind_(density_wind),
-      etot_wind_(etot_wind),
-      eint_wind_(eint_wind),
-      velocity_wind_(velocity_wind),
-      metal_mass_frac_(metal_mass_frac),
-      initialize_uniform_bfield_(initialize_uniform_bfield),
-      perturb_Nwaves_(perturb_Nwaves),
-      perturb_amplitude_(perturb_amplitude),
-      perturb_min_wavelength_(perturb_min_wavelength),
-      perturb_max_wavelength_(perturb_max_wavelength),
-      perturb_seed_(perturb_seed)
+      subsample_n_(p.value_integer("subsample_n",0)),
+      cloud_radius_(p.value_float("cloud_radius",0.0)),
+      cloud_center_x_(p.value_float("cloud_center_x",0.0)),
+      cloud_center_y_(p.value_float("cloud_center_y",0.0)),
+      cloud_center_z_(p.value_float("cloud_center_z",0.0)),
+      density_cloud_(p.value_float("cloud_density",0.0)),
+      density_wind_(p.value_float("wind_density",0.0)),
+      etot_wind_(p.value_float("wind_total_energy",0.0)),
+      eint_wind_(p.value_float("wind_internal_energy",0.0)),
+      velocity_wind_(p.value_float("wind_velocity",0.0)),
+      metal_mass_frac_(p.value_float("metal_mass_fraction",0.0)),
+      initialize_uniform_bfield_(false),
```

Comment:
What is the purpose of this parameter if `initialize_uniform_bfield_` is set on lines 61-72 based on the presence of a `uniform_bfield` parameter list?

Reply:
good point

Reply:
fixed